runtime/server/catalog.go

```diff
@@ -33,12 +35,33 @@ func (s *Server) ListCatalogEntries(ctx context.Context, req *runtimev1.ListCata
 		return nil, status.Error(codes.Unknown, err.Error())
 	}
 
-	pbs := make([]*runtimev1.CatalogEntry, len(entries))
-	for i, obj := range entries {
-		var err error
-		pbs[i], err = catalogObjectToPB(obj)
-		if err != nil {
-			return nil, status.Error(codes.Unknown, err.Error())
+	var pbs []*runtimev1.CatalogEntry
+	if req.Type == runtimev1.ObjectType_OBJECT_TYPE_METRICS_VIEW {
+		for _, entry := range entries {
+			mv := entry.GetMetricsView()
+			policy, err := s.runtime.ResolveMetricsViewPolicy(auth.GetClaims(ctx).Attributes(), req.InstanceId, mv, entry.UpdatedOn)
+			if err != nil {
+				return nil, err
+			}
+
+			if policy != nil && !policy.HasAccess {
+				continue
+			}
+			newMv := filterDimensions(policy, mv)
+			pb, err := mvCatalogObjectToPB(entry, newMv)
+			if err != nil {
+				return nil, status.Error(codes.Unknown, err.Error())
+			}
+			pbs = append(pbs, pb)
+		}
+	} else {
+		pbs = make([]*runtimev1.CatalogEntry, len(entries))
+		for i, obj := range entries {
+			var err error
+			pbs[i], err = catalogObjectToPB(obj)
+			if err != nil {
+				return nil, status.Error(codes.Unknown, err.Error())
+			}
 		}
 	}
```

Comment:
I think it needs to push the conditional inside the for loop. Because if `req.Type` is empty, `ListCatalogEntries` will return all objects (including metrics views)